### PR TITLE
Sc 64 helpdesk ii

### PR DESCRIPTION
**Pull request by Mupico**
Anforderungen aus ISC-64 Helpdesk Umbauen Phase 2, hat immernoch den doublesubmit bug.

### CODE REVIEW

File: controllers/administration.js [hunk 1]
```diff
@@ -1624,7 +1628,8 @@ router.all('/helpdesk', permissionsHelper.permissionsChecker('HELPDESK_VIEW'), f
         qs: {
             $limit: itemsPerPage,
             $skip: itemsPerPage * (currentPage - 1),
-            $sort: req.query.sort
+            $sort: {order: 1},
```
Comment (adrianjost) [inline controllers/administration.js]: Are you sure about this? Without table sorting may not work.

File: static/scripts/loggedin.js [hunk 2]
```diff
@@ -232,6 +240,41 @@ window.addEventListener('DOMContentLoaded', function() {
     if (!/^((?!chrome).)*safari/i.test(navigator.userAgent)) {
         setupFirebasePush();
     }
+
+    let  feedbackSelector = document.querySelector('#feedbackType');
+    if(feedbackSelector){
+        feedbackSelector.onchange = function(){
+            if(feedbackSelector.value === "problem"){
```
Comment (adrianjost) [inline static/scripts/loggedin.js]: Maybe you can use a json config with some  more generic loops to make the code more readable and easier to maintain?

Comment (Mupico) [reply static/scripts/loggedin.js]: i did, please review, i'm not sure if the for loops i made can be improved (forEach doesn't work)

File: static/scripts/loggedin.js [hunk 3]
```diff
@@ -245,33 +245,24 @@ window.addEventListener('DOMContentLoaded', function() {
...
-                document.getElementById("supposedToHappen").required = true;
-                document.getElementById("role").required = false;
-                document.getElementById("desire").required = false;
-                document.getElementById("benefit").required = false;
+                document.getElementById("problemDiv").style.display = "block";
+                document.getElementById("userstoryDiv").style.display = "none";
+                for (let i=0; i < document.getElementById("problemDiv").children.length; i++){
+                    document.getElementById("problemDiv").children[i].children[1].required = true;
+                }
+                for (let i=0; i < document.getElementById("userstoryDiv").children.length; i++){
```
Comment (adrianjost) [inline static/scripts/loggedin.js]:
```js
document.querySelectorAll("#problemDiv input, #problemDiv textarea, #problemDiv select").forEach((node)=>{
   node.required=true;
})
```

**Comment by Mupico** [comment]
Outdated by SC-90, https://github.com/schul-cloud/schulcloud-client/pull/833

**Pull request closed without merging**